### DISCUSSION TITLE
get active custom editor

### DISCUSSION THREAD

**Discussion by safisa** [eclipse-theia/theia#10136] [Q&A] [answered]
Hi,
I working on a vscode extension being used in Theia, and I am consuming a tree view that needs to interact with the active editor that could be a custom editor and not a text editor.
How can I get the active editor (non-text)?
I tried the vscode.window.activeTextEditor API but it is undefined for custom editors !!!

Thanks

**Comment by msujew** [comment, marked as answer]
Hi @safisa, thank you for your question.

Is this strictly related to vscode extensions? As far as I can tell, there's no way in the vscode API to retrieve active non-text editors. However, you're probably better off asking on [stackoverflow with the vscode tag](https://stackoverflow.com/questions/tagged/vscode).

If I'm wrong and vscode does support it, but Theia doesn't, please file a [bug report](https://github.com/eclipse-theia/theia/issues/new?assignees=&labels=&template=bug_report.md) with the appropriate steps to reproduce the issue.

**Reply by safisa** [reply]
Hi,
Yes, there is no API in the vscode extension to retrieve the active non-text editor (e.g a custom editor).
I think I have a workaround for this issue, by having a static member on my custom editor class to maintain the active custom editor, I set it in the resolveCustomEditor method, and on the onDidChangeViewState event (only when visible and active).
Thanks
